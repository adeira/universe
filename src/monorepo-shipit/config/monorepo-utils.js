// @flow strict

import type { ConfigType } from '../ConfigType.flow';

module.exports = ({
  getStaticConfig() {
    return {
      repository: 'git@github.com:adeira/monorepo-utils.git',
    };
  },
  getPathMappings() {
    return new Map([['src/monorepo-utils/', '']]);
  },
}: ConfigType);
