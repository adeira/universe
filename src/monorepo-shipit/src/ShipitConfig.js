// @flow

import os from 'os';
import fs from 'fs';
import path from 'path';
import { warning } from '@adeira/js';

import Changeset from './Changeset';
import addTrackingData from './filters/addTrackingData';
import { commentLines, uncommentLines } from './filters/conditionalLines';
import moveDirectories from './filters/moveDirectories';
import moveDirectoriesReverse from './filters/moveDirectoriesReverse';
import stripExceptDirectories from './filters/stripExceptDirectories';
import stripPaths from './filters/stripPaths';

type ChangesetFilter = (Changeset) => Changeset;

export default class ShipitConfig {
  sourcePath: string;
  destinationPath: string;
  exportedRepoURL: string; // TODO: what to do with this?
  directoryMapping: Map<string, string>;
  strippedFiles: Set<RegExp>;

  #sourceBranch: string;
  #destinationBranch: string;

  constructor(
    sourcePath: string,
    exportedRepoURL: string,
    directoryMapping: Map<string, string>,
    strippedFiles: Set<RegExp>,
    sourceBranch: string = 'origin/master', // our GitLab CI doesn't have master branch
    destinationBranch: string = 'master',
  ) {
    this.sourcePath = sourcePath;
    // This is currently not configurable. We could (should) eventually keep
    // the temp directory, cache it and just update it.
    this.destinationPath = fs.mkdtempSync(path.join(os.tmpdir(), 'adeira-shipit-'));
    this.exportedRepoURL = exportedRepoURL;
    this.directoryMapping = directoryMapping;
    this.strippedFiles = strippedFiles;
    this.#sourceBranch = sourceBranch;
    this.#destinationBranch = destinationBranch;
  }

  getSourceRoots(): Set<string> {
    const roots = new Set();
    for (const root of this.directoryMapping.keys()) {
      warning(fs.existsSync(root) === true, `Directory mapping uses non-existent root: ${root}`);
      roots.add(root);
    }
    return roots;
  }

  getDestinationRoots(): Set<string> {
    // In out cases root is always "". However, if we'd like to export our
    // workspaces to another monorepo then the root would change to something
    // like "my-oss-project/"
    return new Set();
  }

  /**
   * Shipit by default maps directory to match OSS version and strips everything
   * else so we don't publish something outside of the roots scope.
   */
  getDefaultShipitFilter(): ChangesetFilter {
    return (changeset: Changeset) => {
      const ch1 = addTrackingData(changeset);
      const ch2 = stripExceptDirectories(ch1, this.getSourceRoots());
      const ch3 = moveDirectories(ch2, this.directoryMapping);
      const ch4 = stripPaths(ch3, this.strippedFiles);

      // First we comment out lines marked with `@x-shipit-disable`.
      const ch5 = commentLines(ch4, '@x-shipit-disable', '//', null);
      // Then we uncomment lines marked with `@x-shipit-enable`.
      return uncommentLines(ch5, '@x-shipit-enable', '//', null);
    };
  }

  /**
   * Importit reverses the directory mapping and strip some predefined files.
   * It should be in the reversed order from Shipit filters.
   * Please note: there are usually less filters when importing the project (not 1:1 with Shipit).
   */
  getDefaultImportitFilter(): ChangesetFilter {
    return (changeset: Changeset) => {
      // Comment out lines which are only for OSS.
      const ch1 = commentLines(changeset, '@x-shipit-enable', '//', null);
      // Uncomment private code which is disabled in OSS.
      const ch2 = uncommentLines(ch1, '@x-shipit-disable', '//', null);

      const ch3 = stripPaths(ch2, this.strippedFiles);
      return moveDirectoriesReverse(ch3, this.directoryMapping);
    };
  }

  getSourceBranch(): string {
    return this.#sourceBranch;
  }

  getDestinationBranch(): string {
    return this.#destinationBranch;
  }
}
