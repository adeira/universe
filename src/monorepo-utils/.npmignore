__tests__
__flowtests__
BUILD.bazel
BUILD
