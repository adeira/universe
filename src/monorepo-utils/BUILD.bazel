load("//:bazel/macros/build_yarn_workspace.bzl", "build_yarn_workspace")
load("//:bazel/macros/test_yarn_workspace.bzl", "test_yarn_workspace")

build_yarn_workspace(
    name = "monorepo-utils",
    package_name = "@adeira/monorepo-utils",
)

test_yarn_workspace(
    name = "test",
    deps = [
        "@universe//src/js",
    ],
)

# TODO: build `bin` first
# nodejs_binary(
#     name = "monorepo_babel_node",
#     data = [":monorepo_utils"],
#     entry_point = ":bin/monorepo-babel-node.js",
# )
