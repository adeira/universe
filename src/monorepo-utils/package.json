{
  "name": "@adeira/monorepo-utils",
  "description": "Helpful functions to manage monorepos (using Yarn Workspaces).",
  "homepage": "https://github.com/adeira/universe/tree/master/src/monorepo-utils",
  "license": "MIT",
  "private": false,
  "version": "0.2.0",
  "main": "src/index",
  "sideEffects": false,
  "engines": {
    "node": ">=10.10.0"
  },
  "bin": {
    "monorepo-babel-node": "bin/monorepo-babel-node.js",
    "monorepo-run-tests": "bin/monorepo-run-tests.js"
  },
  "dependencies": {
    "@babel/register": "^7.7.7",
    "@adeira/js": "^0.2.0",
    "glob": "^7.1.6",
    "is-ci": "^2.0.0"
  }
}
