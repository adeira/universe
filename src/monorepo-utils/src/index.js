// @flow

export {
  findMonorepoRoot,
  findRootPackageJson,
  findRootPackageJsonPath,
} from './findRootPackageJson';

export { default as ShellCommand } from './ShellCommand';
export { default as ShellCommandResult } from './ShellCommandResult';
export { default as Git } from './Git';
export { default as Workspaces } from './Workspaces';

export { globSync, globAsync } from './glob';

export { default as getTouchedWorkspaces } from './getTouchedWorkspaces';
export { default as getChangedFiles } from './getChangedFiles';
