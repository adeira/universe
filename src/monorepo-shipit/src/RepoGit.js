// @flow strict-local

import fs from 'fs';
import path from 'path';
import { invariant } from '@adeira/js';
import { ShellCommand, ShellCommandResult } from '@adeira/monorepo-utils';

import parsePatch from './parsePatch';
import parsePatchHeader from './parsePatchHeader';
import splitHead from './splitHead';
import Changeset from './Changeset';
import accounts from './accounts';

/**
 * This is our monorepo part - source of exports.
 */
export interface SourceRepo {
  findFirstAvailableCommit(): string;
  findDescendantsPath(
    baseRevision: string,
    headRevision: string,
    roots: Set<string>,
  ): null | $ReadOnlyArray<string>;
  getChangesetFromID(revision: string): Changeset;
  getNativePatchFromID(revision: string): string;
  getNativeHeaderFromIDWithPatch(revision: string, patch: string): string;
  getChangesetFromExportedPatch(header: string, patch: string): Changeset;
}

/**
 * Exported repository containing `adeira-source-id` handles.
 */
export interface DestinationRepo {
  findLastSourceCommit(roots: Set<string>): null | string;
  renderPatch(changeset: Changeset): string;
  commitPatch(changeset: Changeset): string;
  checkoutBranch(branchName: string): void;
  push(branch: string): void;
}

interface AnyRepo {
  // Will most probably always return '4b825dc642cb6eb9a060e54bf8d69288fbee4904'. What? https://stackoverflow.com/q/9765453/3135248
  getEmptyTreeHash(): string;
}

export default class RepoGit implements AnyRepo, SourceRepo, DestinationRepo {
  #localPath: string;

  constructor(localPath: string) {
    invariant(fs.existsSync(path.join(localPath, '.git')), '%s is not a GIT repo.', localPath);
    this.#localPath = localPath;
  }

  _gitCommand: (...args: $ReadOnlyArray<string>) => ShellCommand = (...args) => {
    return new ShellCommand(this.#localPath, 'git', '--no-pager', ...args).setEnvironmentVariables(
      new Map([
        // https://git-scm.com/docs/git#_environment_variables
        ['GIT_CONFIG_NOSYSTEM', '1'],
        ['GIT_TERMINAL_PROMPT', '0'],
      ]),
    );
  };

  push: (destinationBranch: string) => void = (destinationBranch) => {
    this._gitCommand('push', 'origin', destinationBranch).runSynchronously();
  };

  configure: () => void = () => {
    const username = 'adeira-github-bot';
    for (const [key, value] of Object.entries({
      'user.email': accounts.get(username),
      'user.name': username,
    })) {
      // $FlowIssue[incompatible-call]: https://github.com/facebook/flow/issues/2174
      this._gitCommand('config', key, value).runSynchronously();
    }
  };

  // https://git-scm.com/docs/git-checkout
  checkoutBranch: (branchName: string) => void = (branchName) => {
    this._gitCommand(
      'checkout',
      '-B', // create (or switch to) a new branch
      branchName,
    ).runSynchronously();
  };

  clean: () => void = () => {
    this._gitCommand(
      'clean', // remove untracked files from the working tree
      '-x', // ignore .gitignore
      '-f', // force
      '-f', // double force
      '-d', // remove untracked directories in addition to untracked files
    ).runSynchronously();
  };

  isCorrupted: () => boolean = () => {
    const exitCode = this._gitCommand('fsck', '--strict')
      .setNoExceptions()
      .runSynchronously()
      .getExitCode();
    return exitCode !== 0;
  };

  findLastSourceCommit: (roots: Set<string>) => null | string = (roots) => {
    const log = this._gitCommand(
      'log',
      '-1',
      '--grep',
      '^adeira-source-id: \\?[a-z0-9]\\+\\s*$',
      ...roots,
    )
      .setNoExceptions() // empty repo fails with: "your current branch 'master' does not have any commits yet"
      .runSynchronously()
      .getStdout()
      .trim();
    const regex = /adeira-source-id: ?(?<commit>[a-z0-9]+)$/gm;
    let lastCommit = null;
    let match;
    while ((match = regex.exec(log)) !== null) {
      lastCommit = match?.groups?.commit;
    }
    return lastCommit ?? null;
  };

  // https://stackoverflow.com/a/5189296/3135248
  findFirstAvailableCommit: () => string = () => {
    // Please note, the following command may return multiple roots. For example,
    // `git` repository has 6 roots (and we should take the last one).
    const rawOutput = this._gitCommand('rev-list', '--max-parents=0', 'HEAD')
      .runSynchronously()
      .getStdout();
    const rootRevisions = rawOutput.trim().split('\n');
    return rootRevisions[rootRevisions.length - 1];
  };

  getNativePatchFromID: (revision: string) => string = (revision) => {
    return this._gitCommand(
      'format-patch',
      '--no-renames',
      '--no-stat',
      '--stdout',
      '--full-index',
      '--format=', // contain nothing but the code changes
      '-1',
      revision,
    )
      .runSynchronously()
      .getStdout();
  };

  getNativeHeaderFromIDWithPatch: (revision: string, patch: string) => string = (
    revision,
    patch,
  ) => {
    const fullPatch = this._gitCommand(
      'format-patch',
      '--no-renames',
      '--no-stat',
      '--stdout',
      '--full-index',
      '-1',
      revision,
    )
      .runSynchronously()
      .getStdout();
    if (patch.length === 0) {
      // this is an empty commit, so everything is the header
      return fullPatch;
    }
    return fullPatch.replace(patch, '');
  };

  getChangesetFromID: (revision: string) => Changeset = (revision) => {
    const patch = this.getNativePatchFromID(revision);
    const header = this.getNativeHeaderFromIDWithPatch(revision, patch);
    const changeset = this.getChangesetFromExportedPatch(header, patch);
    return changeset.withID(revision);
  };

  getChangesetFromExportedPatch: (header: string, patch: string) => Changeset = (header, patch) => {
    const changeset = parsePatchHeader(header);
    const diffs = new Set();
    for (const hunk of parsePatch(patch)) {
      const diff = this.parseDiffHunk(hunk);
      if (diff !== null) {
        diffs.add(diff);
      }
    }
    return changeset.withDiffs(diffs);
  };

  parseDiffHunk: (hunk: string) => null | { +body: string, +path: string } = (hunk) => {
    const [head, tail] = splitHead(hunk, '\n');
    const match = head.match(/^diff --git [ab]\/(?:.*?) [ab]\/(?<path>.*?)$/);
    if (!match) {
      return null;
    }
    const path = match.groups?.path;
    invariant(path != null, 'Cannot parse path from the hunk.');
    return { path, body: tail };
  };

  // TODO: originally `findNextCommit` - pls reconsider
  findDescendantsPath: (
    baseRevision: string,
    headRevision: string,
    roots: Set<string>,
  ) => null | $ReadOnlyArray<string> = (baseRevision, headRevision, roots) => {
    const log = this._gitCommand(
      'log',
      '--reverse',
      '--ancestry-path',
      '--no-merges',
      '--pretty=tformat:%H',
      `${baseRevision}..${headRevision}`,
      '--', // separates paths from revisions (so you can use non-existent paths)
      ...roots,
    )
      .runSynchronously()
      .getStdout();
    const trimmedLog = log.trim();
    return trimmedLog === '' ? null : trimmedLog.split('\n');
  };

  commitPatch: (changeset: Changeset) => string = (changeset) => {
    if (changeset.getDiffs().size === 0) {
      // This is an empty commit, which `git am` does not handle properly.
      this._gitCommand(
        'commit',
        '--allow-empty',
        '--author',
        changeset.getAuthor(),
        '--date',
        changeset.getTimestamp(),
        '--message',
        changeset.getCommitMessage(),
      ).runSynchronously();
    } else {
      const diff = this.renderPatch(changeset);
      try {
        this._gitCommand('am', '--keep-non-patch', '--keep-cr').setStdin(diff).runSynchronously();
      } catch (error) {
        this._gitCommand('am', '--show-current-patch').setOutputToScreen().runSynchronously();
        this._gitCommand('am', '--abort').setOutputToScreen().runSynchronously();
        throw error;
      }
    }
    // git rev-parse --verify HEAD
    // git --no-pager log -1 --pretty=format:%H
    return this._gitCommand('rev-parse', '--verify', 'HEAD').runSynchronously().getStdout().trim();
  };

  /**
   * Renders changeset to be later used with `git am` command.
   */
  renderPatch: (changeset: Changeset) => string = (changeset) => {
    let renderedDiffs = '';
    const diffs = changeset.getDiffs();
    invariant(diffs.size > 0, 'It is not possible to render empty commit.'); // https://stackoverflow.com/a/34692447

    for (const diff of diffs) {
      const path = diff.path;
      renderedDiffs += `diff --git a/${path} b/${path}\n${diff.body}`;
    }

    // Insert a space before patterns that will make `git am` think that a line in the commit
    // message is the start of a patch, which is an artifact of the way `git am` tries to tell
    // where the message ends and the diffs begin. This fix is a hack; a better fix might be to
    // use `git apply` and `git commit` directly instead of `git am`. It's inspired by the same
    // fix in `facebook/fbshipit` code.
    //
    // See: https://git-scm.com/docs/git-am/2.32.0#_discussion
    // See: https://github.com/git/git/blob/ebf3c04b262aa27fbb97f8a0156c2347fecafafb/mailinfo.c#L649-L683
    // See: https://github.com/facebook/fbshipit/blob/bd0df15c3c18a6645da7a765789ab60c5ffc3a45/src/shipit/repo/ShipItRepoGIT.php#L236-L240
    const commitMessage = changeset
      .getCommitMessage()
      .replace(/^(?<patch>diff -|Index: |---(?:\s\S|\s*$))/m, ' $1');

    // Mon Sep 17 is a magic date used by format-patch to distinguish from real mailboxes
    // see: https://git-scm.com/docs/git-format-patch
    return `From ${changeset.getID()} Mon Sep 17 00:00:00 2001
From: ${changeset.getAuthor()}
Date: ${changeset.getTimestamp()}
Subject: [PATCH] ${commitMessage}

${renderedDiffs}
--
2.21.0
`;
  };

  /**
   * This function exports specified roots from the monorepo. It takes a
   * snapshot of HEAD revision and exports it to the destination path.
   * Please note: this export is unfiltered.
   */
  export: (exportedRepoPath: string, roots: Set<string>) => ShellCommandResult = (
    exportedRepoPath,
    roots,
  ) => {
    const archivePath = path.join(exportedRepoPath, 'archive.tar.gz');
    this._gitCommand(
      'archive',
      '--format=tar',
      `--output=${archivePath}`,
      'HEAD', // TODO
      ...roots,
    ).runSynchronously();
    // Previously, we used only STDIN but that didn't work for some binary files like images for some reason.
    // So now we create an actual archive and use this instead.
    return new ShellCommand(exportedRepoPath, 'tar', '-xvf', archivePath).runSynchronously();
  };

  getEmptyTreeHash(): string {
    return this._gitCommand('hash-object', '-t', 'tree', '/dev/null')
      .runSynchronously()
      .getStdout()
      .trim();
  }
}
