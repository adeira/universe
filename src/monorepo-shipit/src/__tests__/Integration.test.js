// @flow

import { findMonorepoRoot } from '@adeira/monorepo-utils';

import RepoGit from '../RepoGit';
import ShipitConfig from '../ShipitConfig';

// This is a high level integration test which does approximately the same as Shipit with small
// exceptions (for example, we do not try to apply the changeset because we would need to whole
// exported project history). It uses real commits from `adeira/universe` so you can run:
//
// ```
// git show cc7b3818e06f95c2732f75f165a2b98c6eeab135
// ```
//
// This test snapshots various states throughout the export process so you can observe how are the
// filters being applies and how are the changesets being rendered.
test.each([
  // TODO: add more commits as you go
  ['cc7b3818e06f95c2732f75f165a2b98c6eeab135', new Map([['src/eslint-config-adeira/', '']])],
  ['f1cc780d89183b4304ae5c22536baec6ef9736ba', new Map([['src/abacus/', '']])],
])(
  'creates correct changeset from commit %s, filters it and renders it as expected',
  (commitHash, directoryMapping) => {
    const universeRoot = findMonorepoRoot();
    const universeRepo = new RepoGit(universeRoot);

    const changeset = universeRepo.getChangesetFromID(commitHash);

    // First, check that we can construct the changeset as expected:
    expect(changeset).toMatchSnapshot('1 - parsed changeset without filters');

    const shipitConfig = new ShipitConfig(
      universeRoot, // from repo URI
      'mocked.git', // to repo URI
      directoryMapping,
      new Set(), // stripped files
    );
    const defaultFilter = shipitConfig.getDefaultShipitFilter();
    const filteredChangeset = defaultFilter(changeset);

    // Then we check that the default filters were applied to the changeset as expected:
    expect(filteredChangeset).toMatchSnapshot('2 - parsed changeset WITH applied filters');

    // And finally, we verify that the filtered changeset is being rendered as expected:
    expect(universeRepo.renderPatch(filteredChangeset)).toMatchSnapshot('3 - rendered changeset');
  },
);
