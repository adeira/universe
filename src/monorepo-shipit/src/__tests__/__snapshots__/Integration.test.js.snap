// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates correct changeset from commit cc7b3818e06f95c2732f75f165a2b98c6eeab135, filters it and renders it as expected: 1 - parsed changeset without filters 1`] = `
Changeset {
  "author": "\\"dependabot[bot]\\" <49699333+dependabot[bot]@users.noreply.github.com>",
  "coAuthorLines": Array [],
  "debugMessages": Array [],
  "description": "Bumps [prettier](https://github.com/prettier/prettier) from 2.3.0 to 2.3.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.3.0...2.3.1)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>",
  "diffs": Set {
    Object {
      "body": "index 33c6b476374a38f54e7aa3a92d4f9aed68c34624..dd10ce35969c86ef2c5bd8c59e58bb1fea865a58 100644
--- a/src/eslint-config-adeira/package.json
+++ b/src/eslint-config-adeira/package.json
@@ -26,7 +26,7 @@
     \\"eslint-plugin-react-native\\": \\"^3.11.0\\",
     \\"eslint-plugin-relay\\": \\"^1.8.2\\",
     \\"eslint-plugin-sx\\": \\"^0.11.0\\",
-    \\"prettier\\": \\"^2.3.0\\"
+    \\"prettier\\": \\"^2.3.1\\"
   },
   \\"devDependencies\\": {
     \\"deep-diff\\": \\"^1.0.2\\",
",
      "path": "src/eslint-config-adeira/package.json",
    },
    Object {
      "body": "index dd99feca88cc2c4f0561b8b2274d547f4194251a..3dfbae6752edf8d4390293bc3b3c8edb7b26bf56 100644
--- a/src/sx-jest-snapshot-serializer/package.json
+++ b/src/sx-jest-snapshot-serializer/package.json
@@ -11,7 +11,7 @@
   \\"dependencies\\": {
     \\"@adeira/sx\\": \\"^0.25.0\\",
     \\"@babel/runtime\\": \\"^7.14.0\\",
-    \\"prettier\\": \\"^2.3.0\\",
+    \\"prettier\\": \\"^2.3.1\\",
     \\"pretty-format\\": \\"^27.0.2\\"
   },
   \\"devDependencies\\": {
",
      "path": "src/sx-jest-snapshot-serializer/package.json",
    },
    Object {
      "body": "index d486c00d03f695c9ff320d25b0a2523ed6330a6a..3a2c4845ffd37e77e97e71af73c97b4913ca6a1a 100644
--- a/src/sx/package.json
+++ b/src/sx/package.json
@@ -19,7 +19,7 @@
     \\"fast-levenshtein\\": \\"^3.0.0\\",
     \\"json-stable-stringify\\": \\"^1.0.1\\",
     \\"mdn-data\\": \\"^2.0.19\\",
-    \\"prettier\\": \\"^2.3.0\\",
+    \\"prettier\\": \\"^2.3.1\\",
     \\"stylis\\": \\"^4.0.10\\"
   },
   \\"peerDependencies\\": {
",
      "path": "src/sx/package.json",
    },
    Object {
      "body": "index 63ed36a5e5c003c175d9d91a2786b8c1746d11cb..ad33f54ed4ad0afbe5a4561502bfdda0823ef574 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -14086,10 +14086,10 @@ prettier-linter-helpers@^1.0.0:
   dependencies:
     fast-diff \\"^1.1.2\\"
 
-prettier@^2.0.1, prettier@^2.3.0:
-  version \\"2.3.0\\"
-  resolved \\"https://registry.yarnpkg.com/prettier/-/prettier-2.3.0.tgz#b6a5bf1284026ae640f17f7ff5658a7567fc0d18\\"
-  integrity sha512-kXtO4s0Lz/DW/IJ9QdWhAf7/NmPWQXkFr/r/WkR3vyI+0v8amTDxiaQSLzs8NBlytfLWX/7uQUMIW677yLKl4w==
+prettier@^2.0.1, prettier@^2.3.1:
+  version \\"2.3.1\\"
+  resolved \\"https://registry.yarnpkg.com/prettier/-/prettier-2.3.1.tgz#76903c3f8c4449bc9ac597acefa24dc5ad4cbea6\\"
+  integrity sha512-p+vNbgpLjif/+D+DwAZAbndtRrR0md0MwfmOVN9N+2RgyACMT+7tfaRnT+WDPkqnuVwleyuBIG2XBxKDme3hPA==
 
 prettier@~2.2.1:
   version \\"2.2.1\\"
",
      "path": "yarn.lock",
    },
  },
  "id": "cc7b3818e06f95c2732f75f165a2b98c6eeab135",
  "subject": "Bump prettier from 2.3.0 to 2.3.1",
  "timestamp": "Mon, 7 Jun 2021 05:12:25 +0000",
}
`;

exports[`creates correct changeset from commit cc7b3818e06f95c2732f75f165a2b98c6eeab135, filters it and renders it as expected: 2 - parsed changeset WITH applied filters 1`] = `
Changeset {
  "author": "\\"dependabot[bot]\\" <49699333+dependabot[bot]@users.noreply.github.com>",
  "coAuthorLines": Array [],
  "debugMessages": Array [
    "ADD TRACKING DATA: \\"adeira-source-id: cc7b3818e06f95c2732f75f165a2b98c6eeab135\\"",
  ],
  "description": "Bumps [prettier](https://github.com/prettier/prettier) from 2.3.0 to 2.3.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.3.0...2.3.1)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

adeira-source-id: cc7b3818e06f95c2732f75f165a2b98c6eeab135",
  "diffs": Set {
    Object {
      "body": "index 33c6b476374a38f54e7aa3a92d4f9aed68c34624..dd10ce35969c86ef2c5bd8c59e58bb1fea865a58 100644
--- a/package.json
+++ b/package.json
@@ -26,7 +26,7 @@
     \\"eslint-plugin-react-native\\": \\"^3.11.0\\",
     \\"eslint-plugin-relay\\": \\"^1.8.2\\",
     \\"eslint-plugin-sx\\": \\"^0.11.0\\",
-    \\"prettier\\": \\"^2.3.0\\"
+    \\"prettier\\": \\"^2.3.1\\"
   },
   \\"devDependencies\\": {
     \\"deep-diff\\": \\"^1.0.2\\",
",
      "path": "package.json",
    },
  },
  "id": "cc7b3818e06f95c2732f75f165a2b98c6eeab135",
  "subject": "Bump prettier from 2.3.0 to 2.3.1",
  "timestamp": "Mon, 7 Jun 2021 05:12:25 +0000",
}
`;

exports[`creates correct changeset from commit cc7b3818e06f95c2732f75f165a2b98c6eeab135, filters it and renders it as expected: 3 - rendered changeset 1`] = `
"From cc7b3818e06f95c2732f75f165a2b98c6eeab135 Mon Sep 17 00:00:00 2001
From: \\"dependabot[bot]\\" <49699333+dependabot[bot]@users.noreply.github.com>
Date: Mon, 7 Jun 2021 05:12:25 +0000
Subject: [PATCH] Bump prettier from 2.3.0 to 2.3.1

Bumps [prettier](https://github.com/prettier/prettier) from 2.3.0 to 2.3.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.3.0...2.3.1)

 ---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

adeira-source-id: cc7b3818e06f95c2732f75f165a2b98c6eeab135

diff --git a/package.json b/package.json
index 33c6b476374a38f54e7aa3a92d4f9aed68c34624..dd10ce35969c86ef2c5bd8c59e58bb1fea865a58 100644
--- a/package.json
+++ b/package.json
@@ -26,7 +26,7 @@
     \\"eslint-plugin-react-native\\": \\"^3.11.0\\",
     \\"eslint-plugin-relay\\": \\"^1.8.2\\",
     \\"eslint-plugin-sx\\": \\"^0.11.0\\",
-    \\"prettier\\": \\"^2.3.0\\"
+    \\"prettier\\": \\"^2.3.1\\"
   },
   \\"devDependencies\\": {
     \\"deep-diff\\": \\"^1.0.2\\",

--
2.21.0
"
`;

exports[`creates correct changeset from commit f1cc780d89183b4304ae5c22536baec6ef9736ba, filters it and renders it as expected: 1 - parsed changeset without filters 1`] = `
Changeset {
  "author": "=?UTF-8?q?Martin=20Zl=C3=A1mal?= <mrtnzlml@gmail.com>",
  "coAuthorLines": Array [],
  "debugMessages": Array [],
  "description": "",
  "diffs": Set {
    Object {
      "body": "index 23e30baaaa8e3453263fc15c174d4b11ab3a0a6b..5ddd96afcdf20ad5dc36f140d0caa7baaacfb63c 100644
--- a/src/abacus/src/warp_graphql/models.rs
+++ b/src/abacus/src/warp_graphql/models.rs
@@ -24,7 +24,7 @@ pub(in crate::warp_graphql) async fn get_current_user(
                         }
                         User::AnonymousUser(_) => {
                             tracing::error!(\\"Unmatched session token üõë\\");
-                            Err(String::from(\\"Session token doesn't mach any user.\\"))
+                            Err(String::from(\\"Session token doesn't match any user.\\"))
                         }
                         User::AuthorizedUser(user) => {
                             tracing::debug!(\\"Using authorized user: {} üëç\\", user.id());
",
      "path": "src/abacus/src/warp_graphql/models.rs",
    },
  },
  "id": "f1cc780d89183b4304ae5c22536baec6ef9736ba",
  "subject": "Abacus: fix error message typo",
  "timestamp": "Thu, 20 May 2021 16:11:37 -0500",
}
`;

exports[`creates correct changeset from commit f1cc780d89183b4304ae5c22536baec6ef9736ba, filters it and renders it as expected: 2 - parsed changeset WITH applied filters 1`] = `
Changeset {
  "author": "=?UTF-8?q?Martin=20Zl=C3=A1mal?= <mrtnzlml@gmail.com>",
  "coAuthorLines": Array [],
  "debugMessages": Array [
    "ADD TRACKING DATA: \\"adeira-source-id: f1cc780d89183b4304ae5c22536baec6ef9736ba\\"",
  ],
  "description": "adeira-source-id: f1cc780d89183b4304ae5c22536baec6ef9736ba",
  "diffs": Set {
    Object {
      "body": "index 23e30baaaa8e3453263fc15c174d4b11ab3a0a6b..5ddd96afcdf20ad5dc36f140d0caa7baaacfb63c 100644
--- a/src/warp_graphql/models.rs
+++ b/src/warp_graphql/models.rs
@@ -24,7 +24,7 @@ pub(in crate::warp_graphql) async fn get_current_user(
                         }
                         User::AnonymousUser(_) => {
                             tracing::error!(\\"Unmatched session token üõë\\");
-                            Err(String::from(\\"Session token doesn't mach any user.\\"))
+                            Err(String::from(\\"Session token doesn't match any user.\\"))
                         }
                         User::AuthorizedUser(user) => {
                             tracing::debug!(\\"Using authorized user: {} üëç\\", user.id());
",
      "path": "src/warp_graphql/models.rs",
    },
  },
  "id": "f1cc780d89183b4304ae5c22536baec6ef9736ba",
  "subject": "Abacus: fix error message typo",
  "timestamp": "Thu, 20 May 2021 16:11:37 -0500",
}
`;

exports[`creates correct changeset from commit f1cc780d89183b4304ae5c22536baec6ef9736ba, filters it and renders it as expected: 3 - rendered changeset 1`] = `
"From f1cc780d89183b4304ae5c22536baec6ef9736ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Zl=C3=A1mal?= <mrtnzlml@gmail.com>
Date: Thu, 20 May 2021 16:11:37 -0500
Subject: [PATCH] Abacus: fix error message typo

adeira-source-id: f1cc780d89183b4304ae5c22536baec6ef9736ba

diff --git a/src/warp_graphql/models.rs b/src/warp_graphql/models.rs
index 23e30baaaa8e3453263fc15c174d4b11ab3a0a6b..5ddd96afcdf20ad5dc36f140d0caa7baaacfb63c 100644
--- a/src/warp_graphql/models.rs
+++ b/src/warp_graphql/models.rs
@@ -24,7 +24,7 @@ pub(in crate::warp_graphql) async fn get_current_user(
                         }
                         User::AnonymousUser(_) => {
                             tracing::error!(\\"Unmatched session token üõë\\");
-                            Err(String::from(\\"Session token doesn't mach any user.\\"))
+                            Err(String::from(\\"Session token doesn't match any user.\\"))
                         }
                         User::AuthorizedUser(user) => {
                             tracing::debug!(\\"Using authorized user: {} üëç\\", user.id());

--
2.21.0
"
`;
